File: perl_sample/sample01.pl
#
# use -I to specify module(*.pm) directory ex.:
#   perl -I../perl_deploy sample01.pl
#   perl -I. sample01.pl
#
# vim: set et sw=4 sts=4 ai : 

# use utf8;
use strict;
use warnings;

use lib '../perl_deploy';

use TreeMatchLib;
useNewTreeClass Tree1 => 'key0', 'key1';

my $target = TreeConstruct('Tree1', 'A >B (C > D) E')->Tree();
$target->TreePrint();

if (my $result = TreeMatch($target, 'A > . C > .##x'))
{
    print "MATCH\n";
    print $result->Capture('x')->Node()->Attr0() . "\n";
}

File: js_sample/sample01.mjs
'use strict';

import {
    // TreeWrapperBase,
    // TreeWrapperBaseIterator,
    // MatchCapturePlace,
    // TreeMatchLib,
    useNewTreeClass,
    TreeConstruct,
    TreeMatch,
    TreeMatchFind
} from '../js_deploy/TreeMatchLib.mjs';

// (import TreeMatchLib.js)

var Tree1 = useNewTreeClass('Tree1', '__k0', '__k1');

///////////////////////////////////////////////////////////////
function print_last(s)
{
    console.log(s.replace(/\n$/, ''));
}
///////////////////////////////////////////////////////////////

let target = TreeConstruct(Tree1, 'A >B (C > D) E').Tree();
target.TreePrint(print_last);

let result;
if (result = TreeMatch(target, 'A > . C > .##x'))
{
    print_last("MATCH\n");
    print_last(result.Capture('x').Node().Attr0() + "\n");
}

///////////////////////////////////////////////////////////////

